### DISCUSSION TITLE
Funzione rapida per Selezionare linee contigue (adiacenti) iterativamente

### DISCUSSION THREAD

**Discussion by Riccardocasoli** [opendatasicilia/HfcQGIS-md#71] [Domande] [answered]
Sto cercando un modo per selezionare agevolmente tutte le linee adiacenti ad una linea di partenza selezionata manualmente.
Serve in caso di reticolo idrografico o stradale con elementi lineari topologicamente connessi per selezionare interi tratti di corsi d'acqua o segmenti stradali interconnessi: per cui sarebbe utile aggiungere anche la possibilità di filtrare la selezione degli elementi adiacenti  in base alla corrispondenza con il contenuto di un campo indicato in una finestra di dialogo (p.esempio "nome fiume" o "nome strada").
La funzione creata potrebbe essere associata anche ad un comando da tastiera o ad un pulsante personalizzato.
Qualsiasi contributo è apprezzato. Grazie

**Comment by pigreco** [comment, marked as answer]
Ciao Riccardo,
il `Seleziona Elementi per Valore` e il `Seleziona Elementi con Espressione` è gia presente in QGIS (vedi screen sotto, [qui ](https://docs.qgis.org/3.22/it/docs/user_manual/introduction/general_tools.html#automatic-selection)link alla guida): 
![image](https://user-images.githubusercontent.com/7631137/179393150-200baeb2-459a-49c1-8cfe-10c785b6cdbb.png)

Per quanto riguarda la selezione di tutte le linee adiacenti (topologicamente connesse) ad una linea di partenza selezionata manualmente, credo si possa fare usando le potenti espressioni di QGIS.
Appena posso faccio qualche test e condivido le soluzioni :-)

**Reply by pigreco** [reply]
Per selezionare tutte le aste di uno stesso fiume (o strada), a partire da QGIS 3.26 Buones Aires, ci sono due belle novità:

1. se categorizzassi in funzione del nome del fiume, basterebbe utilizzare il menu contestuale nella legenda stessa e selezionare `Seleziona Elementi`; 
![image](https://user-images.githubusercontent.com/7631137/179398477-16d075d6-d4c0-44b5-b545-b7ccba4b41c1.png)

2. utilizzando il comando `Informazioni Elementi` ![](https://docs.qgis.org/3.22/en/_images/mActionIdentify.png) 
![image](https://user-images.githubusercontent.com/7631137/179398555-b2843fbf-6ce0-471b-969f-2fc021bb36be.png)

https://user-images.githubusercontent.com/7631137/179398915-6faa6031-8200-494e-9134-b53a83a307f8.mp4

3. infine, tramite `Selezione per Espressione` utilizzando la seguente espressione:

```
array_agg(
expression:="fiume",
filter:=is_selected())[0]= "fiume"
```
ecco la tabella attributi:
![image](https://user-images.githubusercontent.com/7631137/179398747-7a13b344-1232-42e4-8ff2-960a124c42ca.png)

https://user-images.githubusercontent.com/7631137/179398826-cdab8c66-5997-42ab-ae64-e3a7e76a9a86.mp4

infine, manca solo il caso in cui volessi selezionare le aste a partire SOLO da condizioni geometriche!!!

**Reply by Riccardocasoli** [reply]
...e se volessi inserire un'etichetta sola ad una serie di tratti dello stesso corso d'acqua mantenendo i segmenti separati e tenedo conto che possono esserci rii con nome uguale?
![immagine](https://user-images.githubusercontent.com/6648782/179559122-08a5b21c-6971-4b07-b70c-f0c6540b445c.png)

**Reply by Riccardocasoli** [reply]
<,,,infine, manca solo il caso in cui volessi selezionare le aste a partire SOLO da condizioni geometriche!!!>
in tal caso, scusa la mia ignoranza, qual funzione conviene usare?

**Comment by Riccardocasoli** [comment]
Totò sei un grande! Ti ringrazio tantissimo e non solo per le tue rapide e efficacissime risposte! Provvedo subito a sdebitarmi almeno un pochino dando il mio contributo a HFCQGIS. Ciao. Riccardo

**Reply by pigreco** [reply]
Ciao Riccardo,
se ritieni opportuno, pigia su `Mark as answer` così gli altri possono vedere che hai trovato la soluzione per il tuo quesito.

![image](https://user-images.githubusercontent.com/7631137/179465804-5f641e0e-66eb-4292-9bf2-7ad278bbddad.png)

**Comment by Riccardocasoli** [comment]
...e se volessi inserire un'etichetta sola ad una serie di tratti dello stesso corso d'acqua mantenendo i segmenti separati e tenedo conto che possono esserci rii con nome uguale?
![immagine](https://user-images.githubusercontent.com/6648782/179559122-08a5b21c-6971-4b07-b70c-f0c6540b445c.png)

**Reply by pigreco** [reply]
Buongiorno,
per le etichette esistono molte opzioni e/o configurazioni possibili, devi provarle e vedere quale si adatta ala tuo caso;
a mali estremi estremi rimedi: per gruppi di aste con stesso nome potremmo, tramite condizioni, raggruppare la geometria e far visualizzare unica label (solo in visualizzazione),

Mi rendo conto che non ti ho aiutato molto, ma l'etichettatura è veramente complessa.

PS: per le prossime volta, apri domande singole, non usare una stessa issue per varie domande.
